src/Component/Symfony/EventListener/FormListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Symfony\EventListener;
+
+use Sylius\Component\Resource\Context\Initiator\RequestContextInitiator;
+use Sylius\Component\Resource\Metadata\Operation\HttpOperationInitiator;
+use Sylius\Component\Resource\Symfony\Form\Factory\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+
+final class FormListener
+{
+    public function __construct(
+        private HttpOperationInitiator $operationInitiator,
+        private RequestContextInitiator $contextInitiator,
+        private FormFactoryInterface $formFactory,
+    ) {
+    }
+
+    public function onKernelView(ViewEvent $event): void
+    {
+        $controllerResult = $event->getControllerResult();
+        $request = $event->getRequest();
+        $context = $this->contextInitiator->initializeContext($request);
+        $operation = $this->operationInitiator->initializeOperation($request);
+
+        if (
+            $controllerResult instanceof Response ||
+            null === $operation ||
+            null == $operation->getFormType()
+        ) {
+            return;
+        }
+
+        $form = $this->formFactory->create($operation, $context, $controllerResult);
+        $form->handleRequest($request);
```

Comment:
I still believe this logic should be part of controller, but I don't have resources to refactor it, so I should probably remain silent